homeassistant/components/flume/sensor.py

```diff
@@ -135,72 +138,69 @@ def name(self):
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self._state
+        sensor_key = self._flume_query_sensor[0]
+        if sensor_key not in self._flume_device.values:
+            _LOGGER.debug(
+                "Updating sensor: '%s', key: '%s' is not yet available.",
+                self._name,
+                sensor_key,
+            )
+            return
+
+        _LOGGER.debug(
+            "Updating sensor: '%s', value: '%s'",
+            self._name,
+            self._flume_device.values[sensor_key],
+        )
+        return _format_state_value(self._flume_device.values[sensor_key])
 
     @property
     def unit_of_measurement(self):
         """Return the unit the value is expressed in."""
         # This is in gallons per SCAN_INTERVAL
         return self._flume_query_sensor[1]["unit_of_measurement"]
 
-    @property
-    def available(self):
-        """Device is available."""
-        return self._available
-
     @property
     def unique_id(self):
         """Flume query and Device unique ID."""
         return f"{self._flume_query_sensor[0]}_{self._device_id}"
 
-    def update(self):
-        """Get the latest data and updates the states."""
-
-        def format_state_value(value):
-            return round(value, 1) if isinstance(value, Number) else None
-
-        self._flume_data.update()
-        self._state = format_state_value(
-            self._flume_data.flume_device.values[self._flume_query_sensor[0]]
-        )
-        _LOGGER.debug(
-            "Updating sensor: '%s', value: '%s'",
-            self._name,
-            self._flume_data.flume_device.values[self._flume_query_sensor[0]],
-        )
-        self._available = self._flume_data.available
-
     async def async_added_to_hass(self):
         """Request an update when added."""
+        await super().async_added_to_hass()
         # We do ask for an update with async_add_entities()
```

Comment:
this comment seems wrong maybe "we do not"

Reply:
This is existing, but yes the comment is wrong.  Fixed!